src/uct/sm/mm/mm_posix.c

```diff
@@ -31,11 +31,60 @@ static ucs_config_field_t uct_posix_pd_config_table[] = {
   {NULL}
 };
 
+static ucs_status_t uct_posix_test_mem(size_t length, int shm_fd)
+{
+    int *buf;
+    int i, iters, chunk_size = 256 * 1024;
+    size_t size_to_write, remaining, single_write, total_write = 0;
+
+    buf = (int*) ucs_malloc(length, "write buffer");
+    memset(buf, 0, length);
+
+    if (length > chunk_size) {
```

Comment:
it can be written simpler - just loop while remaining>0 and keep doing remaining-=size_to_write